lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -30,48 +31,61 @@ const ignoredPathComponents = [
   '**/*-renderer.js',
 ];
 
-// @ts-ignore - @types/esprima lacks all of these
-function computeDescription(ast, property, value, startRange) {
-  const endRange = property.range[0];
-
-  for (const comment of ast.comments || []) {
-    if (comment.range[0] < startRange) continue;
-    if (comment.range[0] > endRange) continue;
-    if (comment.value.includes('@')) {
-      // This is a complex description with description and examples.
-      let description = '';
-      /** @type {Record<string, string>} */
-      const examples = {};
-
-      const r = /@(\w+) ({.+})?(.*)(\n|$)/g;
-      let matches;
-      while ((matches = r.exec(comment.value)) !== null) {
-        const tagName = matches[1];
-        const example = matches[2];
-        const messageOrPlaceholder = matches[3].trim();
-
-        if (tagName === 'description') {
-          description = messageOrPlaceholder;
-        } else if (tagName === 'example') {
-          examples[messageOrPlaceholder] = example.substring(1, example.length - 1);
-        }
+/**
+ * Extract the description and examples (if any) from a jsDoc annotation.
+ * @param {import('typescript').JSDoc} ast
+ * @param {string} message
+ * @return {{description: string, examples: Record<string, string>}}
+ */
+function computeDescription(ast, message) {
+  if (ast.tags) {
+    // This is a complex description with description and examples.
+    let description = '';
+    /** @type {Record<string, string>} */
+    const examples = {};
+
+    for (const tag of ast.tags) {
+      const comment = coerceToSingleLineAndTrim(tag.comment);
+
+      if (tag.tagName.text === 'description') {
```

Comment:
Should this throw if we encounter a non-standard tag?

Reply:
That's fair.  I'm not sure I'm convinced, but I guess there isn't much harm in letting new tags roam free.